spark/src/main/java/org/apache/iceberg/spark/source/Reader.java

```diff
@@ -188,6 +167,27 @@ public StructType readSchema() {
     return lazyType();
   }
 
+  /**
+   * This is called in the Spark Driver when data is to be materialized into [ColumnarBatch]s
+   */
+  @Override
+  public List<InputPartition<ColumnarBatch>> planBatchInputPartitions() {
+    Preconditions.checkState(enableBatchRead != null && enableBatchRead, "Batched reads not enabled");
```

Comment:
Why would `enableBatchRead` be null? Why not use a `boolean` instead?

Reply:
enableBatchRead is of type `Boolean`. It is lazily initialized in enableBatchRead().

Reply:
Accessing the `enableBatchRead` that is lazily computed creates an order dependency between this and `enableBatchRead()`. I'd rather simplify this and use the method call directly.